File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashmap = {}
        n = len(nums)
        for i in range(n):
            distance = target - nums[i]
            if distance in hashmap : 
                return (hashmap[distance], i)
            else : 
                hashmap[nums[i]] = i
        return ()

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> hashMap; 
        int n = nums.size();
        for (int i = 0; i < n; i++){
            int dist = target - nums[i];
            if (hashMap.count(dist)){
                return {hashMap[dist], i};
            }
            hashMap[nums[i]] = i;
        }
        return {};
    }
};